usr/src/uts/common/brand/lx/os/lx_brand.c

```diff
@@ -2361,9 +2361,11 @@ lx_elfexec(struct vnode *vp, struct execa *uap, struct uarg *args,
 	 * Save off the important properties of the lx executable. The brand
 	 * library will ask us for this data later, when it is ready to set
 	 * things up for the lx executable.
+	 * For statically linked executable, use vaddr from Phdr.
 	 */
 	edp.ed_phdr = (uphdr_vaddr == -1) ? voffset + ehdr.e_phoff :
```

Comment:
Thanks for opening this, it might explain a few issues we've seen in lx!

Would you mind reworking this as a sequence of if/else statements rather than the nested ternaries? 

https://man.omnios.org/style includes
> Nested ternary operators can be confusing and should be avoided if possible.

Thanks!

Reply:
some thing like this maybe

```c
if (uphdr_vaddr == -1) {
    edp.ed_phdr = voffset + ehdr.e_phoff;
} else {
    if (ehdr.e_type == ET_EXEC && interp == NULL) {
        edp.ed_phdr = uphdr_vaddr;
    } else {
        edp.ed_phdr = voffset + uphdr_vaddr;
    }
}
```

Reply:
I think that online is better for understanding since for online emphasis is on assignment to `edp.ed_phdr`, whereas if user several lines condition with `if` - it is not so clear.

Reply:
Hi @gusev-vitaliy this is less a matter of personal opinion than the coding style of the project <https://man.omnios.org/style#Statements>

Reply:
Updated. Could you please take a look again?

Reply:
But this one didn't check for compilation though. Will do it when this stage is passed.